backend/src/apiserver/template/template.go

```diff
@@ -259,3 +344,87 @@ func toPipelineJobRuntimeConfig(apiRuntimeConfig *api.PipelineSpec_RuntimeConfig
 	runtimeConfig.GcsOutputDirectory = apiRuntimeConfig.GetPipelineRoot()
 	return runtimeConfig, nil
 }
+
+func modelToPipelineJobRuntimeConfig(modelRuntimeConfig *model.RuntimeConfig) (*pipelinespec.PipelineJob_RuntimeConfig, error) {
+	if modelRuntimeConfig == nil {
+		return nil, nil
+	}
+	parameters := new(map[string]*structpb.Value)
+	err := json.Unmarshal([]byte(modelRuntimeConfig.Parameters), parameters)
+	if err != nil {
+		return nil, err
+	}
+	runtimeConfig := &pipelinespec.PipelineJob_RuntimeConfig{}
+	runtimeConfig.ParameterValues = *parameters
+	runtimeConfig.GcsOutputDirectory = modelRuntimeConfig.PipelineRoot
+	return runtimeConfig, nil
+}
+
+func modelToCRDTrigger(modelTrigger model.Trigger) (scheduledworkflow.Trigger, error) {
+	crdCronSchedule := scheduledworkflow.CronSchedule{}
+	crdPeriodicSchedule := scheduledworkflow.PeriodicSchedule{}
+	crdTrigger := scheduledworkflow.Trigger{}
+	// CronSchedule and PeriodicSchedule can have at most one being non-empty
+	if modelTrigger.CronSchedule != (model.CronSchedule{}) {
```

Comment:
I used the method provided [here](https://stackoverflow.com/a/28447372/8216546). If we create `crdCronSchedule` inside the if block, it won't be accessible out side it.

Reply:
I know It works but not ideal to create object for if checking purpose, there's a performance lost here, though not significant, so I said it's optional to change the if. 
For `crdCronSchedule`, you don't need access to it outside the if. In case if evaluates to false, you have a waste of object creation for `crdCronSchedule`.